internal/static/traces/config_test.go

```diff
@@ -1940,3 +1940,135 @@ func sortService(cfg *otelcol.Config) {
 	}
 }
+
+func TestOTelConfig2(t *testing.T) {
+	// tests!
+	tt := []struct {
+		name           string
+		cfg            string
+		expectedError  bool
+		expectedConfig string
+	}{
+		{
+			name: "tail sampling config with DNS load balancing",
+			cfg: `
+receivers:
+  jaeger:
+    protocols:
+      grpc:
+remote_write:
+  - endpoint: example.com:12345
+tail_sampling:
+  policies:
+    - type: always_sample
+    - type: string_attribute
+      string_attribute:
+        key: key
+        values:
+          - value1
+          - value2
+load_balancing:
+  receiver_port: 8181
+  routing_key: service
+  exporter:
+    insecure: true
+  resolver:
+    dns:
+      hostname: agent
+      port: 8282
+      interval: 12m
+      timeout: 76s
+`,
+			expectedConfig: `
+receivers:
+  jaeger:
+    protocols:
+      grpc:
+  push_receiver: {}
+  otlp/lb:
+    protocols:
+      grpc:
+        endpoint: "0.0.0.0:8181"
+exporters:
+  otlp/0:
+    endpoint: example.com:12345
+    compression: gzip
+    retry_on_failure:
+      max_elapsed_time: 60s
+  loadbalancing:
+    routing_key: service
+    protocol:
+      otlp:
+        tls:
+          insecure: true
+        endpoint: noop
+        retry_on_failure:
+          max_elapsed_time: 60s
+        compression: none
+    resolver:
+      dns:
+        hostname: agent
+        port: 8282
+        interval: 12m
+        timeout: 76s
+processors:
+  tail_sampling:
+    decision_wait: 5s
+    policies:
+      - name: always_sample/0
+        type: always_sample
+      - name: string_attribute/1
+        type: string_attribute
+        string_attribute:
+          key: key
+          values:
+            - value1
+            - value2
+extensions: {}
+service:
+  pipelines:
+    traces/0:
+      exporters: ["loadbalancing"]
+      processors: []
+      receivers: ["jaeger", "push_receiver"]
+    traces/1:
+      exporters: ["otlp/0"]
+      processors: ["tail_sampling"]
+      receivers: ["otlp/lb"]
+`,
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			var cfg InstanceConfig
+			err := yaml.Unmarshal([]byte(tc.cfg), &cfg)
+			require.NoError(t, err)
+			// check error
+			actualConfig, err := cfg.OtelConfig()
+			if tc.expectedError {
+				assert.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+
+			// convert actual config to otel config
+			otelMapStructure := map[string]interface{}{}
+			err = yaml.Unmarshal([]byte(tc.expectedConfig), otelMapStructure)
+			require.NoError(t, err)
+
+			factories, err := tracingFactories()
+			require.NoError(t, err)
+
+			expectedConfig, err := otelcolConfigFromStringMap(otelMapStructure, &factories)
+			require.NoError(t, err)
+
+			// Exporters/Receivers/Processors in the config's service.Pipelines, as well as
+			// service.Extensions have to be in the same order for them to be asserted as equal.
+			sortService(actualConfig)
+			sortService(expectedConfig)
+
+			assert.Equal(t, *expectedConfig, *actualConfig)
+		})
+	}
+}
```

Comment:
This is a temporary test which should be deleted prior to the PR merge. It is a copy of one of the cases of "TestOTelConfig", and I copied it here so that it's easier to debug. It fails with this error:
```
--- FAIL: TestOTelConfig2 (0.00s)
    --- FAIL: TestOTelConfig2/tail_sampling_config_with_DNS_load_balancing (0.00s)
        /Users/paulintodev/Documents/GitHub/alloy-2/internal/static/traces/config_test.go:2053: 
            	Error Trace:	/Users/paulintodev/Documents/GitHub/alloy-2/internal/static/traces/config_test.go:2053
            	Error:      	Received unexpected error:
            	            	exporters::loadbalancing: address noop: missing port in address
            	Test:       	TestOTelConfig2/tail_sampling_config_with_DNS_load_balancing
FAIL
FAIL	github.com/grafana/alloy/internal/static/traces	1.061s
```

The error comes from the `res.Validate()` inside `otelcolConfigFromStringMap`, but I don't understand why.